### PR TITLE
eCourts Portal import proof of concept

### PR DESCRIPTION
**Pull request by copelco**
# Summary

* Setup interface to install bookmarklet by dragging link from React app too Chrome bookmarks bar
* Create simple ETL to receive bookmarklet POST request
* Create simple parser interface using [Beautiful Soup](https://www.crummy.com/software/BeautifulSoup/bs4/doc/) to parse name, county, and file number from record HTML
* Start test suite using anonymized HTML record

# Bookmarklet local dev

Inspiration: 
* https://codesandbox.io/s/zlgp3?file=/src/App.js
* https://support.start.me/hc/en-us/articles/200964881-Install-and-use-our-bookmarklet
* https://www.crummy.com/software/BeautifulSoup/bs4/doc/

## Run tests

```shell
pytest -x dear_petition/portal/tests
```

## Local development

```shell
python manage.py runserver
npm run start
```

### Add bookmarklet to your bookmarks bar

Log in and drag the "Portal Importer (development)" button to your bookmarks bar.

### Search and import from eCourts Portal

Visit https://portal-nc.tylertech.cloud/Portal/Home/Dashboard/29 and search for
a record. Click bookmark when on record detail page.

Watch output from Django server.

### CODE REVIEW

**Review by robert-w-gries — COMMENTED** [review]
Comments before testing

File: src/bookmarklet/bookmarklet.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+const POST_URL = import.meta.env.MODE === 'development' ? 'http://localhost:8000' : 'https://www.durhamexpunction.org';
```
Comment (robert-w-gries) [inline src/bookmarklet/bookmarklet.js:1]: TODO: Replace with base window.location url

File: dear_petition/portal/etl/constants.py [hunk 2]
```diff
@@ -0,0 +1,4 @@
+DISMISSED_DISPOSITION_METHODS = (
+    "VD-District Dismissals w/o Leave by DA - No Plea Agreement",
+    "VD-Superior Dismissals w/o Leave by DA - No Plea Agreement",
```
[the remark this answers is not in the document]
Comment (copelco) [reply dear_petition/portal/etl/constants.py:3]: I think we could eventually merge the two together, but agree more data is needed to get an idea of the code variations.

File: dear_petition/portal/etl/parsers/case_summary.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import logging
+import re
+
+from .utils import catch_parse_error
+
+
+logger = logging.getLogger(__name__)
+
+COUNTY_COURT_REGEX = re.compile(r"([\w\s]+)(District|Superior) Court")
+FILENO_REGEX = re.compile(r"\d\dCR\d\d\d\d\d\d-\d\d\d")
```
[the remark this answers is not in the document]
Comment (copelco) [reply dear_petition/portal/etl/parsers/case_summary.py:10]: It's a good question. I'm not sure. We need more records to see.

**Pull request merged**